sdk/samples/iot/paho_iot_provisioning_sample.c

```diff
@@ -358,13 +346,28 @@ static void handle_device_registration_status_message(
     }
     else // Unsuccessful assignment (unassigned, failed or disabled states)
     {
+      uint8_t buffer[100];
+      az_span remainder;
+      az_span operation_status_span = AZ_SPAN_FROM_BUFFER(buffer);
+      if (az_result_failed(az_span_i32toa(
+              operation_status_span, register_response->operation_status, &remainder)))
+      {
+        operation_status_span = AZ_SPAN_FROM_STR("UNKNOWN");
+      }
+      else
+      {
+        remainder = az_span_copy_u8(remainder, 0);
```

Comment:
I don't understand what we are trying to achieve here. `remainder` here  could be an empty span if the above `az_span_i32toa` failed.

Reply:
if az_span_i32toa failes, then operation_status_span  is set to UNKNOWN